### PR TITLE
Authlib injector, local auth server and ely by accounts support

### PR DESCRIPTION
**Pull request by maximmasterr**
This pull request introduces:
* authlib injector - replace mojang server with another
* local auth server -  emulate mojang server
* ely.by account support

We should use this authlib injector and local auth server so that people can connect to games hosted using multimc using direct connect (for  example when you use hamachi). Ely by accounts support added, login to account and login to server works

### CODE REVIEW

**Review by Sebastian-byte — DISMISSED** [review]

File: api/logic/minecraft/MinecraftInstance.cpp [hunk 1]
```diff
@@ -914,6 +917,19 @@ shared_qobject_ptr<LaunchTask> MinecraftInstance::createLaunchTask(AuthSessionPt
...
+        auto step = new InjectAuthlib(pptr, &m_injector);
+        if(session->m_accountPtr->loginType() == "dummy")
+            step->setAuthServer(((QString)"http://localhost:%1").arg(localAuthServerPort));
+        else if(session->m_accountPtr->loginType() == "elyby")
+            step->setAuthServer(((QString) "ely.by").arg(localAuthServerPort));
+        process->appendStep(step);
+    }
+
+    // download authlib
+
```
[the remark this answers is not in the document]
Comment (maximmasterr) [reply api/logic/minecraft/MinecraftInstance.cpp:932]: Oh, I just forgot to delete this comment

File: api/logic/AuthServer.cpp [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+              responseBody = "Not found";
+              responseStatusCode = 404;
+            }
+
+            if(responseBody.length() != 0){
+              responseHeaders << ((QString)"Content-Length: %1").arg(responseBody.length());
+            }
+
+
+            tcpSocket->write(((QString)"HTTP/1.1 %1 OK\r\nConnection: keep-alive\r\n").arg(responseStatusCode).toUtf8());
```
[the remark this answers is not in the document]
Comment (maximmasterr) [reply api/logic/AuthServer.cpp]: Done on a14f6f5

[the remark this answers is not in the document]
Comment (AfoninZ) [reply api/logic/AuthServer.cpp]: Technically Reason-Phrases (status strings) are optional and clients can ignore or modify them, as defined in HTTP/1.1 standard. If really necessary, I'd just add a `responseReasonPhrase` QString and set it together with `responseStatusCode`, then use it in line 62.

**Comment by AfoninZ** [comment]
Looks good to me so far, definitely an improvement over what we have now. Auth server emulation > hacks and dummy values on the client side. Waiting for @Sebastian-byte to approve this

**Comment by maximmasterr** [comment]
Ok, it fails windows build. I think i should rewrite getting injector version

**Review by Sebastian-byte — APPROVED** [review]

**Comment by Sebastian-byte** [comment]
We should squash this merge into a single commit.

**Pull request merged**